p2p/net/swarm/swarm_metrics.go

```diff
@@ -0,0 +1,185 @@
+package swarm
+
+import (
+	"context"
+	"errors"
+	"net"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/network"
+
+	ma "github.com/multiformats/go-multiaddr"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const metricNamespace = "libp2p_swarm_"
+
+var (
+	connsOpened = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: metricNamespace + "connections_opened_total",
+			Help: "Connections Opened",
+		},
+		[]string{"dir", "transport", "security", "muxer"},
+	)
+	keyTypes = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: metricNamespace + "key_types_total",
+			Help: "key type",
+		},
+		[]string{"dir", "key_type"},
+	)
+	connsClosed = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: metricNamespace + "connections_closed_total",
```

Comment:
```suggestion
            Namespace: metricNamespace,
			Name: "connections_closed_total",
```

Likewise with the others.

Reply:
Done.